server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/RemoveRegexMappingRoutesTest.java

```diff
@@ -0,0 +1,131 @@
+
+package org.apache.james.webadmin.routes;
+
+import static io.restassured.RestAssured.with;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import org.apache.james.core.Domain;
+import org.apache.james.core.User;
+import org.apache.james.dnsservice.api.DNSService;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.domainlist.memory.MemoryDomainList;
+import org.apache.james.rrt.lib.Mapping;
+import org.apache.james.rrt.lib.MappingSource;
+import org.apache.james.rrt.memory.MemoryRecipientRewriteTable;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.restassured.RestAssured;
+import io.restassured.filter.log.LogDetail;
+import io.restassured.http.ContentType;
+
+class RemoveRegexMappingRoutesTest {
+
+    private WebAdminServer webAdminServer;
+    private MemoryRecipientRewriteTable memoryRecipientRewriteTable;
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        DNSService dnsService = mock(DNSService.class);
+        DomainList domainList = new MemoryDomainList(dnsService);
+        memoryRecipientRewriteTable = new MemoryRecipientRewriteTable();
+        memoryRecipientRewriteTable.setDomainList(domainList);
+        domainList.addDomain(Domain.of("domain.tld"));
+
+        webAdminServer = WebAdminUtils
+            .createWebAdminServer(new RemoveRegexMappingRoutes(memoryRecipientRewriteTable))
+            .start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(RemoveRegexMappingRoutes.BASE_PATH)
+            .log(LogDetail.METHOD)
+            .build();
+    }
+
+    @AfterEach
+    void stop() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void removeRegexMappingShouldReturnNoContentWhenSuccess() throws Exception {
+        MappingSource mappingSource = MappingSource.fromUser(User.fromUsername("abc@domain.tld"));
+        memoryRecipientRewriteTable
+            .addRegexMapping(mappingSource, "^[aeiou]");
+
+        with()
+            .delete("abc@domain.tld/targets/^[aeiou]")
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204)
+            .contentType(ContentType.JSON);
+
+        assertThat(memoryRecipientRewriteTable.getStoredMappings(mappingSource))
+            .doesNotContain(Mapping.regex("^[aeiou]"));
+    }
+
+    @Test
+    void removeRegexMappingShouldAllowUserWithoutDomain() throws Exception {
+        MappingSource mappingSource = MappingSource.fromUser(User.fromUsername("abcdomaintld"));
+        memoryRecipientRewriteTable
+            .addRegexMapping(mappingSource, "^[aeiou]");
+
+        with()
+            .delete("abcdomaintld/targets/^[aeiou]")
```

Comment:
What if my regex contains an invalid HTTP character ? Like a `/`?

Maybe We need to URL encode this?

Reply:
If a regex contains `/`, it makes the request wrong and then, like when source or regex or both of them are emty, 404 will be returned.

Reply:
My example is not a good one but the principle still stands:

```
Characters that are reserved within a query component and/or have special meaning within a URI/URL:

  reserved    = ";" | "/" | "?" | ":" | "@" | "&" | "=" | "+" | "$" | ","
```

What happens if your URL contains any of the following characters?

URL encoding would be a simple workaround for this issue...